### DISCUSSION TITLE
Is it possible to compile the new version to libretro library?

### DISCUSSION THREAD

**Discussion by RogerYe** [flyinghead/flycast#759] [Q&A]
Thanks to flyinghead's amazing work 

I found the code here is more efficiency than the version in libretro/flycast.
On my board, I got the results :  60 fps   VS   21 fps 

By reading source code, I try to compare the two version , but progess slowly, part of reason is my poor lumbar ^_^
Port this version to the libretro style may be help

Any help will be appreciated

**Comment by flyinghead** [comment]
I don't really understand your question.

Standalone Flycast and the libretro core are using the exact same code. Unless you're running a very old version of the flycast core.

But you didn't specify basic information such as: flycast version, platform, OS, hardware, etc.

**Reply by RogerYe** [reply]
Sorry, maybe I should ask: How to compile the source code in flyinghead/flycast to get a flycast_libretro_android.so ?

**Reply by RogerYe** [reply]
OK. I found the switch in CmakeLists.txt:
option(LIBRETRO "Build libretro core" ON)

Then I "Make Project", and got these message:

C/C++: ../../../../../../../core/deps/libretro-common/glsm/glsm.c:2668: error: undefined reference to 'glClientWaitSync'

Is this related to a wrong NDK version?

**Reply by flyinghead** [reply]
Try something like:
`cmake -DCMAKE_TOOLCHAIN_FILE=/path/to/Android/ndk/22.0.7026061/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_NATIVE_API_LEVEL=28 -DLIBRETRO=ON`

Set the correct path to the ndk.

**Reply by RogerYe** [reply]
I compile this project in ubuntu, cpu is amlogic s812,
focus to get a libretro .so for android

1. option(LIBRETRO "Build libretro core" ON)
get message: C/C++: ../../../../../../../core/deps/libretro-common/glsm/glsm.c:2668: error: undefined reference to 'glClientWaitSync'

And I found that 'glClientWaitSync' is about GLES3. but s812 only support GLES2. so I modified CMakeLists.txt:
option(USE_GLES "Use GLES[3] API" OFF)
option(USE_GLES2 "Use GLES2 API" ON)
......................
	# 2022-09-29  if(ANDROID OR USE_GLES)
    if(USE_GLES)
		target_compile_definitions(${PROJECT_NAME} PRIVATE GLES GLES3 HAVE_OPENGLES HAVE_OPENGLES3)
		find_library(GLES3_LIBRARIES NAMES GLESv3 GLESv2 NO_CACHE REQUIRED)
		target_link_libraries(${PROJECT_NAME} PRIVATE ${GLES3_LIBRARIES})
    elseif(USE_GLES2)
                     .......................

Then will get the message: core/deps/libretro-common/include/glsym/rglgen_headers.h:72:10: fatal error: 'GL/gl.h' file not found

So what happened? Thanks for your help
